src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -115,22 +115,34 @@ CHIP_ERROR ScenesServer::Init()
 
     for (auto endpoint : EnabledEndpointsWithServerCluster(Id))
     {
-        // Explicit AttributeValuePairs is mandatory for matter so we force it here, ScenesName is not but it is forced for now
-        // TODO: We currently force SceneNames on but this needs to be modified to read the value generated from Zap instead.
-        uint32_t featureMap  = to_underlying(Feature::kExplicit) | to_underlying(Feature::kSceneNames);
-        EmberAfStatus status = Attributes::FeatureMap::Set(endpoint, featureMap);
+        uint32_t featureMap;
+        EmberAfStatus status = Attributes::FeatureMap::Get(endpoint, &featureMap);
         if (EMBER_ZCL_STATUS_SUCCESS != status)
         {
-            ChipLogDetail(Zcl, "ERR: setting the scenes FeatureMap on Endpoint %hu Status: %x", endpoint, status);
+            ChipLogDetail(Zcl, "ERR: getting the scenes FeatureMap on Endpoint %hu Status: %x", endpoint, status);
         }
+
+        // Setting NameSupport attribute value to 0x80 if the feature bit is set
         //  The bit of 7 the NameSupport attribute indicates whether or not scene names are supported
         //
         //  According to spec, bit 7 (Scene Names) MUST match feature bit 0 (Scene Names)
-        status = Attributes::NameSupport::Set(endpoint, 0x80);
+        uint8_t nameSupport =
+            (featureMap & to_underlying(Feature::kSceneNames)) ? static_cast<uint8_t>(0x80) : static_cast<uint8_t>(0x00);
```

Comment:
Why are the static cast needed? What are those magic numbers?

Reply:
Ok compiler threats the number as a full int.

 But for the actual value. I meant there is not a bitmap defined for those values?

Reply:
Bitmap will be added in a Zap-cleanup PR.